server/http/oauth2_complete_test.go

```diff
@@ -0,0 +1,270 @@
+package http
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/config"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/mocks"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/user"
+	"github.com/mattermost/mattermost-server/app"
+	"github.com/stretchr/testify/assert"
+)
+
+func getHTTPRequest(userID, rawQuery string) *http.Request {
+	r := &http.Request{
+		Header: make(http.Header),
+	}
+
+	r.URL = &url.URL{
+		RawQuery: rawQuery,
+	}
+	r.Header.Add("Mattermost-User-ID", userID)
+
+	return r
+}
+
+func TestOauth2Complete(t *testing.T) {
+	api := &app.PluginAPI{}
+	kv := &mocks.MockKVStore{
+		KVStore: make(map[string][]byte),
+		Err:     nil,
+	}
+
+	config := &config.Config{}
```

Comment:
1/5 use a go-idiomatic struct literal?

Reply:
I attempted to do that when I was first writing the code. Since the OAuth fields are stored in the StoredConfig which is embedded in the Config struct I am unable to directly assign those in the struct literal. When I do I get the following error:
```
cannot use promoted field StoredConfig.OAuth2Authority in struct literal of type Config
```
We can get around that if the StoredConfig is not embedded in the Config struct.

Reply:
Fair enough, 👍